### DISCUSSION TITLE
Room not set for devices

### DISCUSSION THREAD

**Discussion by metalmatze** [ESPresense/ESPresense-companion#642] [Q&A]
As seen in the JSON payload from running `curl -s http://localhost:8267/api/state/devices | jq '.[0]'` the node in the office can see my devices (which is correct), but the `room` and `floor` are both `null`. Is that because the confidence is `0`? 
```json
{
  "id": "metalmatze_pixel",
  "name": "metalmatze_pixel",
  "nodes": {
    "office": {
      "dist": 1.17,
      "var": null,
      "lh": 11919255.4165
    }
  },
  "room": null,
  "floor": null,
  "confidence": 0,
  "scale": 0.9999942188290866,
  "fixes": 1,
  "lastHit": "2024-05-21T16:58:31.7756275Z",
  "location": {
    "x": 5.165,
    "y": 1.415,
    "z": 0
  },
  "timeout": 30000
}
```
 ## Screenshots 
 
### Map 
It's empty. No devices are shown except when they move. :thinking: 
I would assume the devices would be shown all the time. 

![map](https://github.com/ESPresense/ESPresense-companion/assets/872251/07f45658-0b18-4677-a340-1ed562bf068c)

### Devices

It picks up the X and Y but, again, doesn't actually set the room. I guess, it uses the same API, so that makes sense.

![devices](https://github.com/ESPresense/ESPresense-companion/assets/872251/b9428ac6-4627-4282-bc9b-19c9bdf24bb0)

### Nodes

This seems fine. 
![nodes](https://github.com/ESPresense/ESPresense-companion/assets/872251/63eb7974-340c-4b50-abad-80525ee0e4a0)

### Calibration

It seems like only the ESPresense for the bedroom and living room can see each other. The office ESP is too far away from both.
Is that going to be a problem? 

![calibration](https://github.com/ESPresense/ESPresense-companion/assets/872251/a4d398f5-32b0-414c-9a6e-3e6e23724251)

### ESPresense

Something slightly related but I think unrelated, is that my phone shows up twice (different MAC).
Is this going to cause some hopping between rooms? I've already observed something like that. 

![espresense](https://github.com/ESPresense/ESPresense-companion/assets/872251/8da30cbc-b4fb-46b8-a5f6-2022f6b48a13)

That's it. It's a lot of questions. :smiley: 
Thank you so much for this amazing project! :partying_face:

**Comment by metalmatze** [comment]
Hey!
Anyone have any ideas? Help would be appreciated! Thanks!

**Reply by dalto12593** [reply]
I keep checking all of the discussions and issue threads but there are never replies. I have a lot of the same questions, seems we all do. 

Very sad as this was the most exciting concept out of my entire automation journey.
